aea/test_tools/test_cases.py

```diff
@@ -339,44 +346,77 @@ def replace_file_content(self, src: Path, dest: Path) -> None:
         assert src.is_file() and dest.is_file(), "Source or destination is not a file."
         src.write_text(dest.read_text())
 
+    def _read_out(self, process: subprocess.Popen):
+        for line in TextIOWrapper(process.stdout, encoding="utf-8"):
+            self.stdout[process.pid] += line
 
-class AEAWithOefTestCase(AEATestCase):
-    """Test case for AEA end-to-end tests with OEF node."""
-
-    @pytest.fixture(autouse=True)
-    def _start_oef_node(self, network_node):
-        """Start an oef node."""
-
-    @staticmethod
-    def _read_tty(pid: subprocess.Popen):
-        for line in TextIOWrapper(pid.stdout, encoding="utf-8"):
-            print("stdout: " + line.replace("\n", ""))
-
-    @staticmethod
-    def _read_error(pid: subprocess.Popen):
-        if pid.stderr is not None:
-            for line in TextIOWrapper(pid.stderr, encoding="utf-8"):
-                print("stderr: " + line.replace("\n", ""))
+    def _read_err(self, process: subprocess.Popen):
+        if process.stderr is not None:
+            for line in TextIOWrapper(process.stderr, encoding="utf-8"):
+                self.stderr[process.pid] += line
 
-    def start_tty_read_thread(self, process: subprocess.Popen) -> None:
+    def _start_output_read_thread(self, process: subprocess.Popen) -> None:
         """
-        Start a tty reading thread.
+        Start an output reading thread.
 
         :param process: target process passed to a thread args.
 
         :return: None.
         """
-        self.start_thread(target=self._read_tty, process=process)
+        self.stdout[process.pid] = ""
+        self.start_thread(target=self._read_out, process=process)
 
-    def start_error_read_thread(self, process: subprocess.Popen) -> None:
+    def _start_error_read_thread(self, process: subprocess.Popen) -> None:
         """
         Start an error reading thread.
 
         :param process: target process passed to a thread args.
 
         :return: None.
         """
-        self.start_thread(target=self._read_error, process=process)
+        self.stderr[process.pid] = ""
+        self.start_thread(target=self._read_err, process=process)
+
+    def missing_from_output(
+        self,
+        process: subprocess.Popen,
+        strings: Tuple[str],
+        timeout: int = DEFAULT_PROCESS_TIMEOUT,
+        period: int = 1,
+    ) -> List[str]:
+        """
+        Check if strings are present in process output.
+        Read process stdout in thread and terminate when all strings are present
+        or timeout expired.
+
+        :param process: agent subprocess.
+        :param strings: tuple of strings expected to appear in output.
+        :param timeout: int amount of seconds before stopping check.
+        :param period: int period of checking.
+
+        :return: list of missed strings.
+        """
+        self._start_output_read_thread(process)
```

Comment:
You know, whilst we are at it. It might be helpful to start the error read also. And then in case of some missing strings, print it :)

Reply:
So after line 411 we'd print `stderr` if there are missing strings, otherwise not

Reply:
It's not guaranteed that a missing string coincides with an error, but it would be good to see the output